src/App.tsx

```diff
@@ -1,26 +1,272 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Errors } from './utils/Errors/Errors';
+import { Footer } from './utils/Footer/Footer';
+import { Header } from './utils/Header/Header';
+import { TodoList } from './utils/TodoList/TodoList';
+import { deleteTodos, getTodos, patchTodos, postTodos } from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 700;
+
+enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<string>(Filter.All);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [allCompleted, setAllCompleted] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  // const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const completedTodos = todos.filter(todo => todo.completed).length;
+  const notCompletedTodos = todos.filter(todo => !todo.completed).length;
+  const areAllCompleted =
+    todos?.length > 0 && todos?.every(todo => todo.completed);
+
+  // #region get
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'));
+  }, []);
+
+  const getVisibleTodos = (generalTodos: Todo[], generalFilter: string) => {
+    let filteredTodos = generalTodos;
+
+    if (generalFilter === Filter.Active) {
+      filteredTodos = filteredTodos.filter(todo => !todo.completed);
+    } else if (generalFilter === Filter.Completed) {
+      filteredTodos = filteredTodos.filter(todo => todo.completed);
+    }
```

Comment:
```suggestion
    if (generalFilter === Filter.Active) {
      return  generalTodos.filter(todo => !todo.completed);
    }

    if (generalFilter === Filter.Completed) {
      return generalTodos.filter(todo => todo.completed);
    }
```
you can immediately return and get rid of additional variable and `else if` (separated if blocks are easier to read)